apps/molgenis-components/src/components/forms/InputDecimal.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <FormGroup
+    :id="id"
+    :label="label"
+    :description="description"
+    v-on="$listeners"
+  >
+    <input
+      :id="id"
+      type="number"
+      step="1"
+      :value="value"
+      :class="{ 'form-control': true, 'is-invalid': errorMessage }"
+      :aria-describedby="id + 'Help'"
+      :placeholder="placeholder"
+      @keypress="keyhandler"
+      @input="$emit('input', $event.target.value)"
+    />
+  </FormGroup>
+</template>
+
+<script>
+import BaseInput from "./BaseInput.vue";
+import FormGroup from "./FormGroup.vue";
+import constants from "../constants.js";
+
+const { CODE_0, CODE_9, CODE_BACKSPACE, CODE_DELETE } = constants;
+
+export default {
+  extends: BaseInput,
+  components: {
+    FormGroup,
+  },
+  props: {
+    parser: {
+      default() {
+        return parseFloat;
+      },
+    },
+    errorMessage: { type: String, default: null },
+  },
+  methods: {
+    keyhandler(event) {
+      if (!this.isValidKey(event)) event.preventDefault();
+    },
+    isValidKey(event) {
+      const keyCode = event.which ? event.which : event.keyCode;
```

Comment:
zero is a falsy value , could which have the value zero, and does that mean the keyCode should be used ?

Reply:
There is no key that results in zero, so this should be ok.